modules/miner/poolmanager.go

```diff
@@ -0,0 +1,292 @@
+package miner
+
+import (
+	"errors"
+	"fmt"
```

Comment:
I left in all my testing statements because it's too much to remove them every time I make a commit and the pool is definitely still in testing.